src/components/views/settings/ChangeAvatar.js

```diff
@@ -85,16 +82,15 @@ module.exports = React.createClass({
             newUrl = url;
             if (this.props.onAvatar) {
                 this.props.onAvatar(url);
-            }
-            if (this.props.setAvatar) {
+            } else {
                 if (self.props.room) {
                     return MatrixClientPeg.get().sendStateEvent(
                         self.props.room.roomId,
                         'm.room.avatar',
                         {url: url},
                         ''
                     );
-                } else {
+                } else if (!this.props.groupId) {
```

Comment:
This seems a bit subtle. I'm now thinking that the logic to send state events or set room avatars should be done in similar `onUploadComplete` handlers in the 2(?) other places  that <ChangeAvatar> is used and then at least the API for ChangeAvatar is slightly less subtle. WDYT?